<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>mungo.genomescan</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">
  <!-- Home link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="mungo-module.html">Home</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

      <th class="navbar" width="100%"></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="mungo-module.html">Package&nbsp;mungo</a> ::
        Module&nbsp;genomescan
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="mungo.genomescan-module.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== MODULE DESCRIPTION ==================== -->
<h1 class="epydoc">Module genomescan</h1><p class="nomargin-top"><span class="codelink"><a href="mungo.genomescan-pysrc.html">source&nbsp;code</a></span></p>
<p>genomescan module</p>

<!-- ==================== FUNCTIONS ==================== -->
<a name="section-Functions"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Functions</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Functions"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="mungo.genomescan-module.html#genomescan" class="summary-sig-name">genomescan</a>(<span class="summary-sig-arg">dna</span>,
        <span class="summary-sig-arg">protein</span>,
        <span class="summary-sig-arg">description</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string"></code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">email</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string"></code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">oFileHandle</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">proxy</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">wehiproxy.alpha.wehi.edu.au</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">proxyPort</span>=<span class="summary-sig-default">3128</span>,
        <span class="summary-sig-arg">debug</span>=<span class="summary-sig-default">False</span>)</span><br />
      Returns:
      HTML output of GenomeScan webserver.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="mungo.genomescan-pysrc.html#genomescan">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="mungo.genomescan-module.html#extractSeq" class="summary-sig-name">extractSeq</a>(<span class="summary-sig-arg">feature</span>,
        <span class="summary-sig-arg">blastDb</span>,
        <span class="summary-sig-arg">dx</span>,
        <span class="summary-sig-arg">dy</span>)</span><br />
      Extract the translated sequence of a feature and DNA sequence of the 
      surrounding the genomic region.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="mungo.genomescan-pysrc.html#extractSeq">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="mungo.genomescan-module.html#genomescanFromFeature" class="summary-sig-name">genomescanFromFeature</a>(<span class="summary-sig-arg">feature</span>,
        <span class="summary-sig-arg">blastDb</span>,
        <span class="summary-sig-arg">dx</span>=<span class="summary-sig-default">15000</span>,
        <span class="summary-sig-arg">dy</span>=<span class="summary-sig-default">7000</span>,
        <span class="summary-sig-arg">oFileHandle</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">debug</span>=<span class="summary-sig-default">False</span>)</span><br />
      Predict a gene in the neighbourhood of a known feature.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="mungo.genomescan-pysrc.html#genomescanFromFeature">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="mungo.genomescan-module.html#parseGenomeScanOutput" class="summary-sig-name">parseGenomeScanOutput</a>(<span class="summary-sig-arg">html</span>,
        <span class="summary-sig-arg">debug</span>=<span class="summary-sig-default">False</span>)</span><br />
      A simple stateful parser for genomescan output.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="mungo.genomescan-pysrc.html#parseGenomeScanOutput">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="mungo.genomescan-module.html#parseGenscan" class="summary-sig-name">parseGenscan</a>(<span class="summary-sig-arg">lines</span>,
        <span class="summary-sig-arg">skipLines</span>=<span class="summary-sig-default">2</span>)</span><br />
      Parse lines in genscan or genomescan output.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="mungo.genomescan-pysrc.html#parseGenscan">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
</table>
<!-- ==================== VARIABLES ==================== -->
<a name="section-Variables"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Variables</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Variables"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="__package__"></a><span class="summary-name">__package__</span> = <code title="'mungo'"><code class="variable-quote">'</code><code class="variable-string">mungo</code><code class="variable-quote">'</code></code>
    </td>
  </tr>
</table>
<!-- ==================== FUNCTION DETAILS ==================== -->
<a name="section-FunctionDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Function Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-FunctionDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="genomescan"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">genomescan</span>(<span class="sig-arg">dna</span>,
        <span class="sig-arg">protein</span>,
        <span class="sig-arg">description</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string"></code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">email</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string"></code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">oFileHandle</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">proxy</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">wehiproxy.alpha.wehi.edu.au</code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">proxyPort</span>=<span class="sig-default">3128</span>,
        <span class="sig-arg">debug</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="mungo.genomescan-pysrc.html#genomescan">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>dna</code></strong> - DNA sequence in fasta format.</li>
        <li><strong class="pname"><code>protein</code></strong> - protein sequence in fasta format.</li>
        <li><strong class="pname"><code>description</code></strong> - Description of job (default: '').</li>
        <li><strong class="pname"><code>email</code></strong> - Email address (default: '').</li>
        <li><strong class="pname"><code>oFileHandle</code></strong> - Output file or filename (default: None/no output)</li>
        <li><strong class="pname"><code>proxy</code></strong> - Proxy setting (default: WEHI proxy)</li>
        <li><strong class="pname"><code>proxyPort</code></strong> - Proxy port setting (default: WEHI proxy port)</li>
        <li><strong class="pname"><code>debug</code></strong> - Debug output (default: False)</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>HTML output of GenomeScan webserver.</dd>
  </dl>
</td></tr></table>
</div>
<a name="extractSeq"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">extractSeq</span>(<span class="sig-arg">feature</span>,
        <span class="sig-arg">blastDb</span>,
        <span class="sig-arg">dx</span>,
        <span class="sig-arg">dy</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="mungo.genomescan-pysrc.html#extractSeq">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Extract the translated sequence of a feature and DNA sequence of the 
  surrounding the genomic region.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>feature</code></strong> - Feature object. Mandatory attributes: accession, sStart, sEnd.</li>
        <li><strong class="pname"><code>blastDb</code></strong> - Blast database.</li>
        <li><strong class="pname"><code>dx</code></strong> - Length of sequence to extract upstream.</li>
        <li><strong class="pname"><code>dy</code></strong> - Length of sequence to extract downstream.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>tuple of fasta strings (DNA, protein).</dd>
  </dl>
</td></tr></table>
</div>
<a name="genomescanFromFeature"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">genomescanFromFeature</span>(<span class="sig-arg">feature</span>,
        <span class="sig-arg">blastDb</span>,
        <span class="sig-arg">dx</span>=<span class="sig-default">15000</span>,
        <span class="sig-arg">dy</span>=<span class="sig-default">7000</span>,
        <span class="sig-arg">oFileHandle</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">debug</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="mungo.genomescan-pysrc.html#genomescanFromFeature">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Predict a gene in the neighbourhood of a known feature.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>feature</code></strong> - Feature object. Mandatory attributes: accession, sStart, sEnd.</li>
        <li><strong class="pname"><code>blastDb</code></strong> - Blast database.</li>
        <li><strong class="pname"><code>dx</code></strong> - Length of sequence to extract upstream (default: 20000)</li>
        <li><strong class="pname"><code>dy</code></strong> - Length of sequence to extract downstream (default: 7000)</li>
        <li><strong class="pname"><code>oFileHandle</code></strong> - Output file or filename (default: None/no output)</li>
        <li><strong class="pname"><code>debug</code></strong> - Debug output (default: False)</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>HTML output of GenomeScan webserver.</dd>
  </dl>
</td></tr></table>
</div>
<a name="parseGenomeScanOutput"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">parseGenomeScanOutput</span>(<span class="sig-arg">html</span>,
        <span class="sig-arg">debug</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="mungo.genomescan-pysrc.html#parseGenomeScanOutput">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>A simple stateful parser for genomescan output.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>html</code></strong> - Raw HTML output of genomescan webserver.</li>
        <li><strong class="pname"><code>debug</code></strong> - Debug output (default: False)</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>Tuple (annotation, predicted proteins in fasta format).</dd>
  </dl>
</td></tr></table>
</div>
<a name="parseGenscan"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">parseGenscan</span>(<span class="sig-arg">lines</span>,
        <span class="sig-arg">skipLines</span>=<span class="sig-default">2</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="mungo.genomescan-pysrc.html#parseGenscan">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Parse lines in genscan or genomescan output.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>lines</code></strong> - Lines from genscan/genomescan annotation.</li>
        <li><strong class="pname"><code>skipLines</code></strong> - Number of lines to skip at start (default 3).</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">
  <!-- Home link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="mungo-module.html">Home</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

      <th class="navbar" width="100%"></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Tue Apr 24 16:15:00 2012
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
